ui/src/flux/components/FuncArgs.tsx

```diff
@@ -34,38 +34,40 @@ export default class FuncArgs extends PureComponent<Props> {
     const {name: funcName, id: funcID} = func
     return (
       <div className="func-node--tooltip">
-        {funcName === funcNames.JOIN ? (
-          <Join
-            func={func}
-            bodyID={bodyID}
-            declarationID={declarationID}
-            onChangeArg={onChangeArg}
-            declarationsFromBody={declarationsFromBody}
-            onGenerateScript={onGenerateScript}
-          />
-        ) : (
-          func.args.map(({key, value, type}) => (
-            <FuncArg
-              key={key}
-              type={type}
-              argKey={key}
-              value={value}
+        <div className="func-args">
+          {funcName === funcNames.JOIN ? (
+            <Join
+              func={func}
               bodyID={bodyID}
-              funcID={funcID}
-              funcName={funcName}
-              service={service}
-              onChangeArg={onChangeArg}
               declarationID={declarationID}
+              onChangeArg={onChangeArg}
+              declarationsFromBody={declarationsFromBody}
               onGenerateScript={onGenerateScript}
             />
-          ))
-        )}
-        <div className="func-node--buttons">
+          ) : (
+            func.args.map(({key, value, type}) => (
```

Comment:
Could this be moved outside of the jsx?

Reply:
I was thinking about doing it but now that you mention it, it seems pretty funky